### PR TITLE
RT-280: Setup test coverage

### PR DESCRIPTION
**Pull request by dev-digital**
**Changes**
- Include _coverage_ related packages to pip requirements.
- Update GH Actions to call coverage and comment the result
- Update gitignore
- Un-related Syntax warning fix for test file.

### CODE REVIEW

File: .github/workflows/run-tests.yaml [hunk 1]
```diff
@@ -28,6 +28,8 @@ jobs:
       - name: Bring up db container
         run: docker-compose -f docker-compose.yaml up -d
       - name: Run unit tests
-        run: pytest defend_data_capture
+        run: pytest --cache-clear --cov=defend_data_capture defend_data_capture > pytest-coverage.txt
+      - name: Comment coverage
+        uses: coroo/pytest-coverage-commentator@v1.0.2
```
Comment (dev-digital) [inline .github/workflows/run-tests.yaml:33]: It produces quite a verbose comment to my liking. Happy to disable if appropriate.

Comment (NickFitz) [reply .github/workflows/run-tests.yaml:33]: I don't think it matters too much if it's verbose - most of the time it won't be seen, but when we _do_ need to look at it, more is probably better :)

**Review by NickFitz — APPROVED** [review]
Looks OK to me :)

**Comment by dev-digital** [comment]
> Is there a way to just show the total coverage perhaps? Or just total and then can expand something to see all files? :) Am thinking that having every test listed is really going to take up a lot of space as the app grows.

Exactly, hence my above comment.
Yesterday I found this logged as an [issue](https://github.com/coroo/pytest-coverage-commentator/issues/71) on the tool I am using to produce this comment.

There are ways to publish reports(HTML) from these coverage that I have done in Azure for other projects. While searching for equivalent in GH, I found few actions(one of them is what I have used).

I can explore if GH allows user to publish HTML based reports, if you want

**Review by NickFitz — APPROVED** [review]
Nice! 👍

**Pull request merged**